spec/models/test_script_spec.rb

```diff
@@ -0,0 +1,196 @@
+      expect(@scriptfile).to be_valid
+    end
+  end
+
+
+  # update
+  context 'An invalid script file' do
+    before(:each)  do
+      @asst = create(:assignment,
+                     section_due_dates_type: false,
+                     due_date: 2.days.from_now)
+      display_option = %w(do_not_display display_after_submission display_after_collection)
+
+      @validscriptfile = TestScript.create(assignment_id:               @asst.id,
+                                           seq_num:                     1,
+                                           script_name:                 'validscript.sh',
+                                           description:                 'This is a bash script file',
+                                           max_marks:                   5,
+                                           run_by_instructors:          true,
+                                           run_by_students:             true,
+                                           halts_testing:               false,
+                                           display_description:         display_option[0],
+                                           display_run_status:          display_option[1],
+                                           display_marks_earned:        display_option[2],
+                                           display_input:               display_option[0],
+                                           display_expected_output:     display_option[1],
+                                           display_actual_output:       display_option[2])
+
+      @invalidscriptfile = TestScript.create(assignment_id:             @asst.id,
+                                             seq_num:                   2,
+                                             script_name:               'invalidscript.sh',
+                                             description:               'This is a bash script file',
+                                             max_marks:                 5,
+                                             run_by_instructors:        true,
+                                             run_by_students:           true,
+                                             halts_testing:             false,
+                                             display_description:       display_option[2],
+                                             display_run_status:        display_option[1],
+                                             display_marks_earned:      display_option[0],
+                                             display_input:             display_option[2],
+                                             display_expected_output:   display_option[1],
+                                             display_actual_output:     display_option[0])
+    end
+
+    context 'script file expected to be invalid when assignment is nil' do
+      it 'return false when assignment is nii' do
+        @invalidscriptfile.assignment_id = nil
```

Comment:
Please separate the words in the variable name with underscores (i.e., use `snake_case`)